### ISSUE TITLE
Feature: Add Hash Key Partitioning

### ISSUE THREAD

**Issue by itajenglish** [memphisdev/memphis#1242]
### Current behavior

Currently, Memphis only allows producing messages in partitions via a Round Robin method.

### Suggested solution

Allow devs to produce messages using a Hash Partitioning method by specifying a `key` field or leveraging the existing `msgId` field when producing a message to assign specific messages to be consumed in order by a single consumer.

Example:
```
await producer.produce({
    message: 'test'
    key: '12345'
});
```

### Additional context

This is extremely important for systems that must always consume messages produced with the same key in order by the same consumer. This would bring Memphis to same ordering guarantees as Kafka.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

### Contributing Docs

- [ ] I agree to follow this project's Contribution Docs

**Comment by yanivbh1** [comment]
Hey @itajenglish,
Coming up on the next release :)

**Comment by yanivbh1** [comment]
Released on v1.3.0
